dask/tests/test_base.py

```diff
@@ -432,3 +432,11 @@ def test_persist_array_bag():
 
     assert np.allclose(x, xx)
     assert list(b) == list(bb)
+
+
+def test_encoding():
+    """
+    Tests for: Unicode errors on string columns with NaNs #1883
+    """
+    df = pd.DataFrame({'x': [u'foo', u'Jos\xe9', np.nan]})
+    dd.from_pandas(df, npartitions=2)
```

Comment:
Request to cleanup this test. A few things:

- The test suite should run fine without any of the optional dependencies installed (this includes numpy/pandas). You can skip tests that require an optional dependency by using `pytest.mark.skipif('some code string')`. In this case, I'd skip if numpy is missing.
- Since the code change is actually for numpy arrays, request that the input is a numpy array
- Ideally, the name of the test is descriptive enough that you don't need a comment/docstring.
- The failing issue was actually in `tokenize`, which is what you should be testing here. It's helpful to test the minimal components, which makes isolating and debugging failures easier.
- Slight organizational nit - I like to keep like tests together, as such I might move this test up with the other tokenize array tests (not a big deal though).

Putting these together, I might rewrite this as:

```python
@pytest.mark.skipif('not np')
def test_tokenize_object_array_with_nans():
    a = np.array([u'foo', u'Jos\xe9', np.nan])
    assert tokenize(a) == tokenize(a)
```